### PR TITLE
Support deploy with UDC in `deploy` command

### PR DESCRIPTION
**Pull request by Arcticae**
Support for migrator will come in a separate PR (merging into a side-branch)

### CODE REVIEW

File: protostar/commands/deploy_command.py [hunk 1]
```diff
@@ -39,19 +39,26 @@ def description(self) -> str:
...
                 is_positional=True,
             ),
+            ProtostarArgument(
+                name="max-fee",
+                description=(
+                    "The maximum fee that the sender is willing to pay for the transaction. "
+                    'Provide "auto" to auto estimate the fee.'
+                ),
+                type="fee",
+            ),
```
Comment (mkaput) [inline protostar/commands/deploy_command.py:61]: perhaps we could extract this into commons module?

File: tests/e2e/test_gateway_commands.py [hunk 2]
```diff
@@ -30,13 +46,30 @@ def test_deploying_and_interacting_with_contract(
...
         dst="./src/main.cairo",
     )
+
     protostar(["build"])
 
     # TODO(mkaput): Use structured output when it'll be available in `deploy`.
     result = protostar(
         [
-            "deploy",
+            "--no-color",
```
Comment (mkaput) [inline tests/e2e/test_gateway_commands.py:55]: declare has JSON output on `main`, you can use it here

File: protostar/starknet_gateway/gateway_facade.py [hunk 3]
```diff
@@ -100,81 +141,95 @@ def __init__(
     def get_starknet_requests(self) -> List[StarknetRequest]:
         return self._starknet_requests.copy()
 
-    async def deploy(
+    async def deploy_with_udc(
```
Comment (mkaput) [inline protostar/starknet_gateway/gateway_facade.py:144]:
```suggestion
    async def deploy_via_udc(
```

File: protostar/starknet_gateway/gateway_facade.py [hunk 4]
```diff
@@ -78,6 +81,44 @@ class DeployAccountArgs:
...
+
+    cairo_inputs = transform_constructor_inputs_from_python(abi, inputs)
+    validate_cairo_inputs(abi, cairo_inputs)
+    return cairo_inputs
+
+
+def is_cairo_data(inputs: CairoOrPythonData) -> bool:
+    return isinstance(
+        inputs, list
+    )  # Rest of cases should be typesafe, since we don't accept an array of dicts
```
Comment (mkaput) [inline protostar/starknet_gateway/gateway_facade.py:102]:
```suggestion
from typing_extensions import TypeGuard
def is_cairo_data(inputs: CairoOrPythonData) -> TypeGuard[CairoData]:
    return isinstance(inputs, list)
```

File: protostar/starknet_gateway/gateway_facade.py [hunk 5]
```diff
@@ -100,81 +141,95 @@ def __init__(
...
-        cairo_inputs = self._prepare_constructor_inputs(inputs, compiled_contract_path)
-
-        tx = make_deploy_tx(
-            compiled_contract=compiled_contract,
-            constructor_calldata=cairo_inputs,
+        abi: Optional[AbiType] = None,
+        account_address: Optional[int] = None,
+        salt: Optional[int] = None,
+        token: Optional[str] = None,
+    ):
```
Comment (mkaput) [inline protostar/starknet_gateway/gateway_facade.py:155]:
```suggestion
    ) -> SuccessfulDeployResponse:
```

**Review by mkaput — CHANGES_REQUESTED** [review]

File: protostar/commands/deploy_command.py (hunk 1 above)
Comment (Arcticae) [reply protostar/commands/deploy_command.py:61]: Out of scope of this PR

File: protostar/commands/deploy_command.py [hunk 6]
```diff
@@ -39,19 +39,26 @@ def description(self) -> str:
...
     @property
     def arguments(self):
         return [
             BLOCK_EXPLORER_ARG,
             ProtostarArgument(
-                name="contract",
-                description="The path to the compiled contract.",
-                type="path",
-                is_required=True,
+                name="class_hash",
```
Comment (kasperski95) [inline protostar/commands/deploy_command.py:49]: All Protostar arguments use dashes instead of floors.

**Review by mkaput — APPROVED** [review]

**Review by kasperski95 — CHANGES_REQUESTED** [review]

**Review by kasperski95 — APPROVED** [review]

**Pull request merged**